centreon/src/Core/TimePeriod/Infrastructure/API/AddTimePeriod/AddTimePeriodController.php

```diff
@@ -0,0 +1,119 @@
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\Infrastructure\Common\Api\DefaultPresenter;
+use Core\TimePeriod\Application\UseCase\AddTimePeriod\{
+    AddTimePeriod, AddTimePeriodRequest
+};
+use Core\Infrastructure\Common\Api\Router;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+class AddTimePeriodController extends AbstractController
+{
+    /**
+     * @param Request $request
+     * @param AddTimePeriod $useCase
+     * @param AddTimePeriodsPresenter $presenter
+     *
+     * @return object
+     *
+     * @throws AccessDeniedException
+     */
+    public function __invoke(
+        Request $request,
+        AddTimePeriod $useCase,
+        AddTimePeriodsPresenter $presenter,
+        Router $router
+    ): object {
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+        try {
+            /**
+             * @var array{
+             *     name: string,
+             *     alias: string,
+             *     days: array<array{
+             *         day: integer,
+             *         time_range: string
+             *     }>,
+             *     templates: int[],
+             *     exceptions: array<array{
+             *         day_range: string,
+             *         time_range: string
+             *     }>
+             * } $dataSent
+             */
+            $dataSent = $this->validateAndRetrieveDataSent($request, __DIR__ . '/AddTimePeriodSchema.json');
+            $dtoRequest = $this->createDtoRequest($dataSent);
+            $useCase($dtoRequest, $presenter);
+        } catch (\Throwable $ex) {
+            $presenter->setResponseStatus(
+                new ErrorResponse($ex->getMessage())
+            );
+        }
+        return $presenter->show();
+    }
+
+    /**
+     * @param array{
+     *     name: string,
+     *     alias: string,
+     *     days: array<array{
+     *         day: integer,
+     *         time_range: string
+     *     }>,
+     *     templates: int[],
+     *     exceptions: array<array{
+     *         day_range: string,
+     *         time_range: string
+     *     }>
+     * } $dataSent
+     * @return AddTimePeriodRequest
+     */
+    private function createDtoRequest(array $dataSent): AddTimePeriodRequest
+    {
+        $dto = new AddTimePeriodRequest();
+        $dto->name = $dataSent['name'];
+        $dto->alias = $dataSent['alias'];
+        $dto->days = array_map(function (array $day): array {
+            return [
+                'day' => $day['day'],
+                'time_range' => $day['time_range']
+            ];
+        }, $dataSent['days']);
+        $dto->templates = $dataSent['templates'];
+        $dto->exceptions = array_map(function (array $exception): array {
+            return [
+                'day_range' => $exception['day_range'],
+                'time_range' => $exception['time_range'],
+            ];
+        }, $dataSent['exceptions']);
```

Comment:
According to the schema: days, templates and exceptions arrays are optionals and therefore may be omitted in the payload